README.md

```diff
@@ -1,6 +1,69 @@
 # Module Factory
+
 [![Build Status](https://github.com/gnosis/module-factory/actions/workflows/ci.yml/badge.svg)](https://github.com/gnosis/module-factory/actions)
 [![Coverage Status](https://coveralls.io/repos/github/gnosis/module-factory/badge.svg?branch=master)](https://coveralls.io/github/gnosis/module-factory?branch=master)
 
-This repository aims to make easier the deployment and track of Safe Modules using [Minimal Proxy Pattern](https://eips.ethereum.org/EIPS/eip-1167)
+The purpose of the Module Factory repository is to make the deployment of Safe Modules easier. Applying the Minimal Proxy Pattern, this module reduces the gas cost and simplifies the track of deployed modules. The Minimal Proxy Pattern has been used because the modules do not need to be upgradeable since a safe can deploy a new one. It's worth mentioning that it costs roughly 5,000 extra gas for each transaction when you use a proxy, so after a certain amount of transactions (~700) it will, probably, be cheaper to deploy the module from the constructor rather than the proxy.
+
+There's also a JS API, allowing the developers to interact with the ProxyFactory Contract more easily. You can check the factory file to see more details, it consists of 4 methods, which are:
+
+### Deploy and set up module
```

Comment:
```suggestion
### 1. Deploy and set up module
```
I suggest enumerating this because you mentioned how many there would be. Alternatively above you could say something like "4 methods, described individually in the following sections". Maybe it's just me, but the colon indicates a list or enumerated list to follow, rather than a new section header.